code/messaging/src/test/java/com/adobe/marketing/mobile/messaging/internal/MessagingFullscreenMessageDelegateTests.java

```diff
@@ -241,11 +232,33 @@ public void test_overrideUrlLoadWithJavascriptPayload() {
         when(mockMessageSettings.getParent()).thenReturn(internalMessage);
 
         // test
-        internalMessage.overrideUrlLoad(mockFullscreenMessage, "adbinapp://dismiss?js=(function test() { return 'javascript value'; })();");
+        internalMessage.overrideUrlLoad(mockFullscreenMessage, "adbinapp://dismiss?interaction=javascript&amp;link=js%3D%28function%28%29+%7B+return+%27javascript+value%27%3B+%7D%29%28%29%3B");
 
-        // verify encoded javascript evaluated by the webview object
+        // verify javascript evaluated by the webview object
         ArgumentCaptor<String> stringArgumentCaptor = ArgumentCaptor.forClass(String.class);
         verify(mockWebView, times(1)).evaluateJavascript(stringArgumentCaptor.capture(), any(ValueCallback.class));
-        assertEquals("(function test() { return 'javascript value'; })();", stringArgumentCaptor.getValue());
+        assertEquals("js=(function() { return 'javascript value'; })();", stringArgumentCaptor.getValue());
+    }
+
+    @Test
+    public void test_overrideUrlLoadWithDeeplink() {
+        // setup
+        runWithMockedServiceProvider(() -> {
+            try {
+                internalMessage = new InternalMessage(mockMessagingExtension, createRuleConsequence(), new HashMap<>(), new HashMap<>(), mockWebView, mockWebViewHandler, scriptHandlerMap);
+            } catch (Exception exception) {
+                fail(exception.getMessage());
+            }
+            when(mockFullscreenMessage.getMessageSettings()).thenReturn(mockMessageSettings);
+            when(mockMessageSettings.getParent()).thenReturn(internalMessage);
+
+            // test
+            internalMessage.overrideUrlLoad(mockFullscreenMessage, "adbinapp://dismiss?interaction=deeplink&link=scheme%3A%2F%2Fparameters%3Fparam1%3Dvalue1%26param2%3Dvalue2");
```

Comment:
Should `&` be html escaped here `"adbinapp://dismiss?interaction=deeplink&amp;link=...`?

Reply:
html unescaping will be removed as the deeplink is unescaped when passed to overrideUrlLoad